feat: 💪 impl load testing and updated docs / OpenAPI

Diegomcha opened the pull request:
# Summary
> [!NOTE]
> Depends on #138 and #139 

This pull request introduces significant updates to the documentation and the `gatewayservice` API, focusing on new features, enhanced clarity, and improved observability. Key changes include the addition of new diagrams, updates to the OpenAPI specification to support custom game configurations and AI game modes, and improved documentation structure.

## Changes
### Documentation Enhancements:

* **New Diagrams for System Behavior**:
  - Added a detailed sequence diagram for game behavior, covering game initialization, hint requests, and statistics retrieval (`docs/diagrams/6_3_diagram.plantuml`, `docs/diagrams/6_4_diagram.plantuml`). [[1]](diffhunk://#diff-32a39770e8f30df58c4a118844881b56b7fc34bfcd19d639497a8da212667b16R1-R113) [[2]](diffhunk://#diff-a260e5b40b8317ec6874f33c572dfcafcc3d9afda479262b54c4342f7b2cfbe9R1-R29)
  - Updated the runtime view to rename "History of a user" to "Statistics of a user" and clarified the interaction flow (`docs/src/06_runtime_view.adoc`).

* **Structural Updates to Documentation**:
  - Added a new "Test Report" section, detailing testing methodologies, coverage, and findings (`docs/src/12_test_report.adoc`).
  - Rearranged and renamed sections in the main documentation index for better organization (`docs/index.adoc`).

* **Terminology and Context Adjustments**:
  - Updated descriptions in the business and technical context to reflect changes in system behavior, such as the use of AI for hints instead of questions (`docs/src/03_context_and_scope.adoc`). [[1]](diffhunk://#diff-e2d137f69516c834263947f6c891b3b02abdbc825fa59f7020345a94c977f0abL63-R66) [[2]](diffhunk://#diff-e2d137f69516c834263947f6c891b3b02abdbc825fa59f7020345a94c977f0abL99-R99)
  - Clarified the use of a "non-pure microservices pattern" for reduced complexity (`docs/src/04_solution_strategy.adoc`).

### API Updates:

* **Support for Custom Game Configurations**:
  - Added a new `/game/config` endpoint to configure games with custom settings, including game modes, round limits, and AI mode (`gatewayservice/openapi.yaml`). [[1]](diffhunk://#diff-95bec93b2860fdad1a84b75ae038cea45ce52e4cdb5dd7f26c980a6266fb120fR115-R156) [[2]](diffhunk://#diff-95bec93b2860fdad1a84b75ae038cea45ce52e4cdb5dd7f26c980a6266fb120fL658-R694) [[3]](diffhunk://#diff-95bec93b2860fdad1a84b75ae038cea45ce52e4cdb5dd7f26c980a6266fb120fR782-R817)

* **Removed Deprecated Features**:
  - Removed the `/game/quit` endpoint, simplifying the API and focusing on active features (`gatewayservice/openapi.yaml`).

* **Enhanced Game Configuration Schema**:
  - Updated the game configuration schema to support multiple modes and an AI game flag (`gatewayservice/openapi.yaml`). [[1]](diffhunk://#diff-95bec93b2860fdad1a84b75ae038cea45ce52e4cdb5dd7f26c980a6266fb120fR711-R714) [[2]](diffhunk://#diff-95bec93b2860fdad1a84b75ae038cea45ce52e4cdb5dd7f26c980a6266fb120fR782-R817)

### Observability Improvements:

* **Logging and Monitoring**:
  - Introduced a new "Observability" section in the documentation, detailing the use of Grafana, Prometheus, and a consistent logging system to monitor requests and debug issues (`docs/src/08_concepts.adoc`).

### Load testing
Implemented a basic test covering the most important functionalities of the application.

Diegomcha committed:
feat(gateway): :memo: updated OpenAPI documentation

Signed-off-by: Diego <38155922+Diegomcha@users.noreply.github.com>

Diegomcha committed:
docs: general update to the documentation and diagrams

Signed-off-by: Diego <38155922+Diegomcha@users.noreply.github.com>

Diegomcha committed:
docs: :memo: added tests report

Signed-off-by: Diego <38155922+Diegomcha@users.noreply.github.com>

Diegomcha committed:
test: :white_check_mark: added load tests

Signed-off-by: Diego <38155922+Diegomcha@users.noreply.github.com>

Diegomcha committed:
docs: :sparkles: improve documentation

Signed-off-by: Diego <38155922+Diegomcha@users.noreply.github.com>

Diegomcha committed:
test: :zap: included last run of load tests

Signed-off-by: Diego <38155922+Diegomcha@users.noreply.github.com>

Diegomcha committed:
docs: :memo: improved documentation

Signed-off-by: Diego <38155922+Diegomcha@users.noreply.github.com>

Diegomcha committed:
ci: :green_heart: exclude generated report from SonarQube

Signed-off-by: Diego <38155922+Diegomcha@users.noreply.github.com>

Diegomcha committed:
docs(gateway): :memo: updated OpenAPI documentation

Signed-off-by: Diego <38155922+Diegomcha@users.noreply.github.com>

Diegomcha committed:
chore: :twisted_rightwards_arrows: update branch

Signed-off-by: Diego <38155922+Diegomcha@users.noreply.github.com>

Diegomcha committed:
docs: :memo: updated documentation

Signed-off-by: Diego <38155922+Diegomcha@users.noreply.github.com>

Diegomcha committed:
test: :test_tube: update last run load tests report

Signed-off-by: Diego <38155922+Diegomcha@users.noreply.github.com>

Diegomcha committed:
docs: :memo: added documentation on how to run the load tests

Signed-off-by: Diego <38155922+Diegomcha@users.noreply.github.com>

Diegomcha committed:
chore: :twisted_rightwards_arrows: updated branch

Signed-off-by: Diego <38155922+Diegomcha@users.noreply.github.com>

UO288992 commented on gatewayservice/openapi.yaml:
```diff
@@ -655,11 +707,18 @@ components:
+                  - painter
+                  - writer
+              example:
                 - musicians
```
Should be musician, like in the enum field.

Diegomcha replied:
I've pushed a local fix

UO288992 commented on gatewayservice/openapi.yaml:
```diff
@@ -743,14 +806,54 @@ components:
+              - painter
+              - writer
+          example:
+            - musicians
```
Should be musician, like in the enum field.

Diegomcha replied:
I've made a local fix will be pushed in a few minutes

uo288499 committed:
fix: :art: Fixing final sonar qube issues and small change in docs

Diegomcha committed:
fix: :bug: small fixes

Signed-off-by: Diego <38155922+Diegomcha@users.noreply.github.com>

UO288992 reviewed (changes requested):
Minor changes must be made in openapi.

Diegomcha committed:
feat(gateway): :card_file_box: enable grafana and prometheus

Signed-off-by: Diego <38155922+Diegomcha@users.noreply.github.com>

UO288992 committed:
refactor(llm): :coffin: Unused function removed.

uo288499 committed:
docs: :memo: Update some parts of documentation

Functional requirements updated / Added more context to acceptance tests / Some minor chnages in quality requirements / Better maping of container and application

uo288499 committed:
Merge branch 'feat/gw-openapi' of https://github.com/Arquisoft/wichat_en1c into feat/gw-openapi

uo288499 committed:
docs: :memo: Changes to docs and openapi.yaml

Modified worng 6.2 login diagram that said that we could choose avatar / Added the sapnish language support in 02_architecture_constraints / Added question, category and answerAI to question and gameQuestion in openapi.yaml

uo288499 reviewed (approved):
It seems good, the changes done in documentation clarify and correct things up. The changes to the js classes themselves solves the sonarqube issues. The changes to openapi.yaml reflects better the real application. Finally the configuration for load testing seems extense and complete
